prov/gni/test/rdm_rw.c

```diff
@@ -0,0 +1,316 @@
+#include "gnix_rma.h"
+
+#include <criterion/criterion.h>
+
+static struct fid_fabric *fab;
+static struct fid_domain *dom;
+static struct fid_ep *ep[2];
+static struct fid_av *av;
+static struct fi_info *hints;
+static struct fi_info *fi;
+void *ep_name[2];
+size_t gni_addr[2];
+static struct fid_cq *send_cq;
+static struct fi_cq_attr cq_attr;
+
+void rdm_rma_setup(void)
+{
+	int ret = 0;
+	struct fi_av_attr attr;
+	size_t addrlen = 0;
+
+	hints = fi_allocinfo();
+	cr_assert(hints, "fi_allocinfo");
+
+	hints->domain_attr->cq_data_size = 4;
+	hints->mode = ~0;
+
+	hints->fabric_attr->name = strdup("gni");
+
+	ret = fi_getinfo(FI_VERSION(1, 0), NULL, 0, 0, hints, &fi);
+	cr_assert(!ret, "fi_getinfo");
+
+	ret = fi_fabric(fi->fabric_attr, &fab, NULL);
+	cr_assert(!ret, "fi_fabric");
+
+	ret = fi_domain(fab, fi, &dom, NULL);
+	cr_assert(!ret, "fi_domain");
+
+	attr.type = FI_AV_MAP;
+	attr.count = 16;
+
+	ret = fi_av_open(dom, &attr, &av, NULL);
+	cr_assert(!ret, "fi_av_open");
+
+	ret = fi_endpoint(dom, fi, &ep[0], NULL);
+	cr_assert(!ret, "fi_endpoint");
+
+	cq_attr.format = FI_CQ_FORMAT_CONTEXT;
+	cq_attr.size = 1024;
+	cq_attr.wait_obj = 0;
+
+	ret = fi_cq_open(dom, &cq_attr, &send_cq, 0);
+	cr_assert(!ret, "fi_cq_open");
+
+	ret = fi_ep_bind(ep[0], &send_cq->fid, FI_TRANSMIT);
+	cr_assert(!ret, "fi_ep_bind");
+
+	ret = fi_getname(&ep[0]->fid, NULL, &addrlen);
+	cr_assert(addrlen > 0);
+
+	ep_name[0] = malloc(addrlen);
+	cr_assert(ep_name[0] != NULL);
+
+	ep_name[1] = malloc(addrlen);
+	cr_assert(ep_name[1] != NULL);
+
+	ret = fi_getname(&ep[0]->fid, ep_name[0], &addrlen);
+	cr_assert(ret == FI_SUCCESS);
+
+	ret = fi_endpoint(dom, fi, &ep[1], NULL);
+	cr_assert(!ret, "fi_endpoint");
+
+	ret = fi_getname(&ep[1]->fid, ep_name[1], &addrlen);
+	cr_assert(ret == FI_SUCCESS);
+
+	ret = fi_av_insert(av, ep_name[0], 1, &gni_addr[0], 0,
+				NULL);
+	cr_assert(ret == 1);
+
+	ret = fi_av_insert(av, ep_name[1], 1, &gni_addr[1], 0,
+				NULL);
+	cr_assert(ret == 1);
+
+	ret = fi_ep_bind(ep[0], &av->fid, 0);
+	cr_assert(!ret, "fi_ep_bind");
+
+	ret = fi_ep_bind(ep[1], &av->fid, 0);
+	cr_assert(!ret, "fi_ep_bind");
+}
+
+void rdm_rma_teardown(void)
+{
+	int ret = 0;
+
+	ret = fi_close(&ep[0]->fid);
+	cr_assert(!ret, "failure in closing ep.");
+
+	ret = fi_close(&ep[1]->fid);
+	cr_assert(!ret, "failure in closing ep.");
+
+	ret = fi_close(&av->fid);
+	cr_assert(!ret, "failure in closing av.");
+
+	ret = fi_close(&dom->fid);
+	cr_assert(!ret, "failure in closing domain.");
+
+	ret = fi_close(&fab->fid);
+	cr_assert(!ret, "failure in closing fabric.");
+
+	fi_freeinfo(fi);
+	fi_freeinfo(hints);
+	free(ep_name[0]);
+	free(ep_name[1]);
+}
+
+/*******************************************************************************
+ * Test vc functions.
+ ******************************************************************************/
+
+TestSuite(rdm_rma, .init = rdm_rma_setup, .fini = rdm_rma_teardown,
+	  .disabled = false);
+
+#define BUF_SZ 4096
+Test(rdm_rma, rw)
+{
```

Comment:
NICE!

@ztiffany @hppritcha @jswaro @jshimek 
On a related note, should we have the unit tests silent (no output) by default?  Maybe we could use the criterion verbose flag to get more verbose output?  FWIW, in the automated builds, I use `--verbose`, so we would see the output if there is an error.